diff --git a/target/linux/ath79/image/generic-tp-link.mk b/target/linux/ath79/image/generic-tp-link.mk
index fed572c884..1812d26003 100644
--- a/target/linux/ath79/image/generic-tp-link.mk
+++ b/target/linux/ath79/image/generic-tp-link.mk
@@ -734,6 +734,19 @@ define Device/tplink_tl-wr2543-v1
 endef
 TARGET_DEVICES += tplink_tl-wr2543-v1

+define Device/tplink_tl-wr703n
+  $(Device/tplink-16mlzma)
+  SOC := ar9331
+  DEVICE_MODEL := TL-WR703N
+  DEVICE_VARIANT := v1
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2
+  TPLINK_HWID := 0x07030101
+  CONSOLE := ttyATH0,115200
+  IMAGE/factory.bin := append-rootfs | mktplinkfw factory -C US
+  SUPPORTED_DEVICES += tl-wr703n
+endef
+TARGET_DEVICES += tplink_tl-wr703n
+
 define Device/tplink_tl-wr710n-v1
   $(Device/tplink-8mlzma)
   SOC := ar9331
